### PR TITLE
Method any

### PR DESCRIPTION
**Pull request by LuisARodr**
This adds the `any(spec)` method functionality to the framework.
Also includes tests for the method.

Description of the method:
>This method shall return a true value if any of the elements in the array satisfies the given spec.  If the spec is a function then it shall be called for each element in the array, whenever the spec function returns a true value then immediately the any function will success (return a true value), but if it doesn’t for any of the collection elements then the any function will fail (return false value).  If the spec parameter is not a function object, then it will be tested against each element of the array and it will success if any elements matches the spec parameter value.

### CODE REVIEW

File: Methods/ArrayExtension.js [hunk 1]
```diff
@@ -38,3 +38,36 @@ Array.prototype.where = function where(spec) {
 
     return result;
 }
+
+/**
+ * Returns true when the value of spec is on the array, if spec is a function
```
Comment (jesus-castro3) [inline Methods/ArrayExtension.js:43]: What type is `@param spec`?  provide a type and structure.
Also if `any` function returns a value you should also include it.

example: `@return {number}`

See JSDocs for guidance.

File: Methods/ArrayExtension.js [hunk 2]
```diff
@@ -38,3 +38,36 @@ Array.prototype.where = function where(spec) {
...
+ * @param spec A value to look for on the array, if spec is a function then it 
+ * evaluates every value on the array, the function takes two params: the 
+ * current value of the array and the index of the value.
+ */
+Array.prototype.any = function any(spec) {
+    if ((typeof spec) == 'function') {
+        let satisfiesFunction = false;
+        
+        this.each(function evaluate(value, index) {
+            if (spec.call(this, value, index)) {
```
Comment (jesus-castro3) [inline Methods/ArrayExtension.js]: Is any asking only for one spec to be `true`? If yes, then we don't need to run through the whole array, return once we find the first one to be `true`.

File: Methods/ArrayExtension.js [hunk 3]
```diff
@@ -38,3 +38,36 @@ Array.prototype.where = function where(spec) {
...
+        });
+
+        return satisfiesFunction;
+    }
+    
+    
+    let isInside = false;
+
+    this.each(function evaluate(value) {
+        if (value === spec) {
```
Comment (jesus-castro3) [inline Methods/ArrayExtension.js:63]: same here

File: Methods/ArrayExtension.js (hunk 1 above)
Comment (jesus-castro3) [reply Methods/ArrayExtension.js:43]: docs: https://devdocs.io/jsdoc/about-getting-started

File: Methods/ArrayExtension.js (hunk 2 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js]: solved on aa29647

File: Methods/ArrayExtension.js (hunk 3 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js:63]: Also solved on aa29647

File: Methods/ArrayExtension.js (hunk 1 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js:43]: Thanks I'll give it a read to make the changes

Comment (LuisARodr) [reply Methods/ArrayExtension.js:43]: made changes on the `any` method on 79bd0fc,

File: Methods/ArrayExtension.js [hunk 4]
```diff
@@ -38,3 +38,42 @@ Array.prototype.where = function where(spec) {
...
+ * then it evaluates every value in the array and returns true if spec returns
+ * true for any evaluation, otherwise it returns false.
+ * @param {spec|*} spec - Value to look for in the array, if spec is a function then it 
+ * evaluates every value on the array.
+ * @returns {boolean} - True if the value is found on the array or spec is evaluated
+ * as true, false otherwise.
+ */
+Array.prototype.any = function any(spec) {
+    if ((typeof spec) == 'function') {
+        for (let i = 0; i < this.length; i++) {
```
Comment (jesus-castro3) [inline Methods/ArrayExtension.js:53]: this can also be done using the .some method from array, but thsi looks good also

**Review by jesus-castro3 — APPROVED** [review]

**Pull request merged**